theano/misc/check_blas.py

```diff
@@ -143,6 +143,12 @@ def test():
                   help="The numpy memory layout parameter used when creating"
                   " the numpy.ndarray objects. It accepts 'C' for C memory"
                   " order and 'F' for Fortran order (for all matrices).")
+parser.add_option('--small_only', action='store_true', dest='small_only',
+                  default=False,
+                  help="If true, additionally perform gemm on large matrices.")
+parser.add_option('-B', '--B', action='store', dest='B',
+                  default=5000, type="int",
```

Comment:
I like that. But I would make it work a little differently. Can you make M,N and K default to 0. If they are 0, use the value of B, otherwise, use the corresponding value. Then you don't need the small_only parameter.